activemodel/lib/active_model/attribute_methods.rb

```diff
@@ -489,6 +489,13 @@ def respond_to?(method, include_private_methods = false)
       end
     end
 
+    # Provide the pattern matching interface for matching against hash patterns.
```

Comment:
```suggestion
    # Returns a hash of attributes for the given keys. Provides the pattern matching interface for matching against hash patterns.
```

We should also add a code example here.